packages/strapi-plugin-users-permissions/admin/src/components/PopUpForm/index.js

```diff
@@ -8,127 +8,195 @@ import React from 'react';
 import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
 import { FormattedMessage } from 'react-intl';
 import PropTypes from 'prop-types';
-import { get } from 'lodash';
-import { router } from 'app';
+import {
+  capitalize,
+  get,
+  findIndex,
+  isArray,
+  isEmpty,
+  isObject,
+  includes,
+  map,
+  startsWith,
+  tail,
+  take,
+  takeRight,
+} from 'lodash';
+
+// Translations
+import en from 'translations/en.json';
 
 import Input from 'components/Input';
 
 import styles from './styles.scss';
 
 class PopUpForm extends React.Component { // eslint-disable-line react/prefer-stateless-function
-  toggleModal = () => router.push(router.location.pathname);
+  state = { enabled: false, isEditing: false };
 
-  renderButton = () => {
-    if (this.props.showLoader) {
-      return (
-        <Button onClick={() => {}} type="submit" className={styles.primary} disabled>
-          <p className={styles.saving}>
-            <span>.</span><span>.</span><span>.</span>
-          </p>
-        </Button>
-      );
+  getRedirectURIProviderConf = () => { // NOTE: Still testings providers so the switch statement is likely to change
```

Comment:
It should be handled by the backend for the exceptions... If we add a new provider in the future and if we add the ability to add a custom provider for the users, it means he also has to edit the plugin and rebuild it.

Reply:
Well while testing we found out that the redirecURI has to be editable. This function is used only for displaying what the user has to set in his providers application's configurations.

Should I add a comment on this method to reflect that?